### PR TITLE
Update LuksBootKeyFileJob.cpp to check for unencrypted boot with encrypted root

### PR DESCRIPTION
**Pull request by bkmo**
Check to not create luks keyfile with unencrypted boot and encrypted root.

### CODE REVIEW

**Comment by bkmo** [comment]
I didn't touch this as it didn't seem to work anyways:
`for ( const auto& d : s.devices )
    {
        // Skip setupLuks for root partition if system has an unencrypted /boot
        if ( d.isRoot && hasUnencryptedSeparateBoot() )
        {
            continue;
        }`

**Comment by dalto8** [comment]
Isn't the real issue in the prior condition?

In other words, this is what is currently in the PR:
```
    if ( hasUnencryptedSeparateBoot() && !hasEncryptedRoot() )
    {
        // /boot partition is not encrypted, keyfile must not be used
        // But only if root partition is not encrypted
        cDebug() << Logger::SubEntry << "/boot partition is not encrypted, skipping keyfile creation.";
        return Calamares::JobResult::ok();
    }

     if ( hasUnencryptedSeparateBoot() && hasEncryptedRoot() )
    {
        // /boot partition is not encrypted, keyfile must not be used
        // But only if root partition is encrypted
        cDebug() << Logger::SubEntry << "/boot partition is not encrypted, skipping keyfile creation.";
        return Calamares::JobResult::ok();
    }
```

But isn't that the same as:
```
    if ( hasUnencryptedSeparateBoot() )
    {
        // /boot partition is not encrypted, keyfile must not be used
        cDebug() << Logger::SubEntry << "/boot partition is not encrypted, skipping keyfile creation.";
        return Calamares::JobResult::ok();
    }
```

**Comment by bkmo** [comment]
> Isn't the real issue in the prior condition?
> 
> In other words, this is what is currently in the PR:
> 
> ```
>     if ( hasUnencryptedSeparateBoot() && !hasEncryptedRoot() )
>     {
>         // /boot partition is not encrypted, keyfile must not be used
>         // But only if root partition is not encrypted
>         cDebug() << Logger::SubEntry << "/boot partition is not encrypted, skipping keyfile creation.";
>         return Calamares::JobResult::ok();
>     }
> 
>      if ( hasUnencryptedSeparateBoot() && hasEncryptedRoot() )
>     {
>         // /boot partition is not encrypted, keyfile must not be used
>         // But only if root partition is encrypted
>         cDebug() << Logger::SubEntry << "/boot partition is not encrypted, skipping keyfile creation.";
>         return Calamares::JobResult::ok();
>     }
> ```
> 
> But isn't that the same as:
> 
> ```
>     if ( hasUnencryptedSeparateBoot() )
>     {
>         // /boot partition is not encrypted, keyfile must not be used
>         cDebug() << Logger::SubEntry << "/boot partition is not encrypted, skipping keyfile creation.";
>         return Calamares::JobResult::ok();
>     }
> ```

It looks to in fact do the same.

**Comment by bkmo** [comment]
So we can do away with the first condition having unencrypted root as it doesn't matter with the unencrypted boot.

**Comment by dalto8** [comment]
> So we can do away with the first condition having unencrypted root as it doesn't matter with the unencrypted boot.

Since they are both basically the same, you can remove either one as long as you also remove the check of encrypted root and the associated comment.

File: src/modules/luksbootkeyfile/LuksBootKeyFileJob.cpp [hunk 1]
```diff
@@ -294,14 +294,13 @@ LuksBootKeyFileJob::exec()
...
+    if ( hasUnencryptedSeparateBoot() )
     {
-        // /boot partition is not encrypted, keyfile must not be used
-        // But only if root partition is not encrypted
+        // /boot partition is not encrypted, keyfile must not be used.
         cDebug() << Logger::SubEntry << "/boot partition is not encrypted, skipping keyfile creation.";
         return Calamares::JobResult::ok();
     }
-
+
```
Comment (dalto8) [inline src/modules/luksbootkeyfile/LuksBootKeyFileJob.cpp]: Looks like some extra spaces slipped in here.

**Comment by bkmo** [comment]
> > So we can do away with the first condition having unencrypted root as it doesn't matter with the unencrypted boot.
> 
> Since they are both basically the same, you can remove either one as long as you also remove the check of encrypted root and the associated comment.

Ok, I already made the exact change on my end.

**Comment by bkmo** [comment]
> > > So we can do away with the first condition having unencrypted root as it doesn't matter with the unencrypted boot.
> > 
> > 
> > Since they are both basically the same, you can remove either one as long as you also remove the check of encrypted root and the associated comment.
> 
> Ok, I already made the exact change on my end.

I don't know what you were referring to as also a problem with grubcfg

File: src/modules/luksbootkeyfile/LuksBootKeyFileJob.cpp (hunk 1 above)
Comment (dalto8) [reply src/modules/luksbootkeyfile/LuksBootKeyFileJob.cpp]: You marked this as resolved but it still seems to be present.

**Comment by dalto8** [comment]
> I don't know what you were referring to as also a problem with grubcfg

In my testing, grubcfg is enabling cryptodisk even when `/boot` is unencrypted.

**Comment by bkmo** [comment]
> > I don't know what you were referring to as also a problem with grubcfg
> 
> In my testing, grubcfg is enabling cryptodisk even when `/boot` is unencrypted.

`#GRUB_ENABLE_CRYPTODISK=y`   Was added to a clean install with unencrypted boot.

**Comment by dalto8** [comment]
> > > I don't know what you were referring to as also a problem with grubcfg
> > 
> > 
> > In my testing, grubcfg is enabling cryptodisk even when `/boot` is unencrypted.
> 
> `#GRUB_ENABLE_CRYPTODISK=y` Was added to a clean install with unencrypted boot.

Was the root encrypted in addition to /boot being unencrypted?

**Comment by bkmo** [comment]
> Was the root encrypted in addition to /boot being unencrypted?

Yes the root was encrypted.  Now to look into the creation of cryptab.
Looks to be this in encrypt_hook_nokey in initramfs module:
`. /usr/share/initramfs-tools/hook-functions
  if [ -f /etc/crypttab ]
  then
  	cp /etc/crypttab ${DESTDIR}/etc/
  fi`

So if there is an unencrypted boot then we can change the main.py to not do anything instead of the encrypt_hook_nokey, but I am not sure what to do.
` if unencrypted_separate_boot:
            shutil.copy2(
                os.path.join(_path, "encrypt_hook_nokey"),
                target
                )`

**Comment by dalto8** [comment]
The initramfscfg changes should be moved to a separate PR.  However, I am not confident they are correct.  If you open a separate PR for those, be sure to clearly explain why you made those changes and how you tested them.

**Comment by bkmo** [comment]
> The initramfscfg changes should be moved to a separate PR. However, I am not confident they are correct. If you open a separate PR for those, be sure to clearly explain why you made those changes and how you tested them.

I made those changes to test them, not knowing they would be committed. I am sure they are not correct, but am just playing. Now how do I un-commit them?

**Comment by dalto8** [comment]
> Looks to be this in encrypt_hook_nokey in initramfs

The initramfscfg module is not related.  You wouldn't normally be using that module since you are testing on an Arch-based ISO.

**Comment by bkmo** [comment]
> > Looks to be this in encrypt_hook_nokey in initramfs
> 
> The initramfscfg module is not related. You wouldn't normally be using that module since you are testing on an Arch-based ISO.

It is the only place I encountered crypttab, but now I see what you mean.

**Comment by dalto8** [comment]
> Now how do I un-commit them?

Use `git reset --hard` to return to previous commit and then force push your branch.

**Comment by bkmo** [comment]
> > Now how do I un-commit them?
> 
> Use `git reset --hard` to return to previous commit and then force push your branch.

Ok , but I am working from the web gui. I will clone locally and see how it goes.

**Comment by bkmo** [comment]
No luck with deleting the commits. After a hard reset the force push is everything up to date. I will probably just delete the branch and call it a day.

**Comment by kkofler** [comment]
You are not hard-resetting to the correct commit. You have to specify the commit hash for `git reset --hard`.

**Comment by bkmo** [comment]
> You are not hard-resetting to the correct commit. You have to specify the commit hash for `git reset --hard`.

Thanks, I think I have it sorted out now.

**Comment by bkmo** [comment]
I'm done chasing the crypttab creation on unencrypted boot. The only module I found it in was initramfscfg. At least this seem to take care of the keyfile creation.

**Comment by bkmo** [comment]
This is working well but could use a tweak to get rid of a warning on compile. Maybe you can do it for me please  @dalto8 
` LuksBootKeyFileJob.cpp:246:1: warning: ‘bool hasEncryptedRoot()’ defined but not used [-Wunused-function]
  246 | hasEncryptedRoot()`

**Comment by dalto8** [comment]
That means that function isn't used anymore.  You just need to delete that function.

**Comment by bkmo** [comment]
> That means that function isn't used anymore. You just need to delete that function.

I realize that but when I tried I had an error. I'm not a coder.

**Comment by dalto8** [comment]
You need to give me rights to push to your repo.

**Comment by bkmo** [comment]
I added you as a collaborator.

**Comment by bkmo** [comment]
Thank you, the only other issue is there is an entry to crypptab for the keyfile even with unencrypted boot. It is unnecessary in this scenario.

**Pull request merged**